controllers/snapshot/snapshot_adapter.go

```diff
@@ -120,29 +120,14 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (controller.Operation
 					"integrationTestScenario.Name", integrationTestScenario.Name,
 					"len(integrationPipelineRuns)", len(*integrationPipelineRuns))
 			} else {
-				a.logger.Info("Creating new pipelinerun for integrationTestscenario",
-					"integrationTestScenario.Name", integrationTestScenario.Name)
 				pipelineRun, err := a.createIntegrationPipelineRun(a.application, &integrationTestScenario, a.snapshot)
-				if err != nil {
+				if pipelineRun == nil && err != nil {
 					a.logger.Error(err, "Failed to create pipelineRun for snapshot and scenario")
```

Comment:
We may inform which pipelinerun failed as part as prompting error in the log.
Just out of curiosity, asking about the condition:  **pipelineRun == nil && err != nil**
What is the case when pipelinerun is not nil and error is not nil either? Is there such situation?

Reply:
I agree, if there is err != nil, we shouldn't care about return variable at all

Reply:
1. When the pipelineRun is not created successfully, I don't think we need to log it out.
2. When the pipelineRun is created successfully but it fails to update snapshot to `InProgress`, we just log some error but won't `return controller.RequeueWithError(err)`, see [here](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L138-L139).
    When we failed to set ControllerReference or create pipelineRun, we will return `nil, error` and then `return controller.RequeueWithError(err)`

Reply:
2. it still means that we need to check err != nil only in that condition, it cannot happen that we return error and the working pipeline at the same time

Reply:
Good point, there was a check for `err != nil`, but I removed it. Updated. Thanks!